homeassistant/components/omnilogic/__init__.py

```diff
@@ -0,0 +1,217 @@
+"""The Omnilogic integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from omnilogic import LoginException, OmniLogic, OmniLogicException
+
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, PlatformNotReady
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    COORDINATOR,
+    DOMAIN,
+    OMNI_API,
+    POLL_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Omnilogic component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Omnilogic from a config entry."""
+
+    conf = entry.data
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    polling_interval = conf[POLL_INTERVAL]
+    api = OmniLogic(username, password)
+
+    try:
+        await api.connect()
+        await api.get_telemetry_data()
+    except LoginException as error:
+        _LOGGER.debug("OmniLogic login error: %s", error)
+        raise PlatformNotReady from error
+    except OmniLogicException as error:
+        _LOGGER.debug("OmniLogic API error: %s", error)
```

Comment:
Is this error just ignored?

Reply:
Will add a PlatformNotReady issue on this error.

Reply:
Should be `ConfigEntryNotReady`